storage/store.go

```diff
@@ -1587,20 +1602,21 @@ func (s *Store) processRaft() {
 					panic(err) // TODO(bdarnell)
 				}
 			}
+			s.processRaftMu.Unlock()
```

Comment:
I looked around and couldn't find anything in go to specify the dependency
of the locking.

On Tue, Feb 2, 2016 at 10:50 AM Ben Darnell <notifications@github.com>
wrote:

> In storage/store.go
> <https://github.com/cockroachdb/cockroach/pull/4076#discussion_r51586550>:
>
> > @@ -1587,20 +1602,21 @@ func (s *Store) processRaft() {
> >  					panic(err) // TODO(bdarnell)
> >  				}
> >  			}
> > +			s.processRaftMu.Unlock()
>
> This isn't introducing any potential deadlock that wasn't there already,
> because there was the same relationship between the goroutine itself and
> mu. I don't know of any way to enforce this in go but I'd love to see
> some annotations and an analyzer like
> http://clang.llvm.org/docs/ThreadSafetyAnalysis.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/4076/files#r51586550>.
>